Assets/UnityIO.cs

```diff
@@ -0,0 +1,48 @@
+using UnityEngine;
+using System;
+using System.IO;
+using System.Collections;
+
+public class UnityIO
+{
+	public class Response {
+		public string error;
+		public byte[] data;
+
+		public Response(string e, byte[] d) {
+			error = e;
+			data = d;
+		}
+		public Response(string e, string d) {
+			error = e;
+			data = Convert.FromBase64String(d);
+		}
+	}
+
+	public delegate void IORequestCallback(object userData, Response response);
+
+	public IEnumerator FetchData(Uri uri, IORequestCallback callback, object userData) {
```

Comment:
Partially addressed this using lambdas. Need to see how an interface could be a better approach to handle this. But lambda usage makes this look much better.